### ISSUE TITLE
Detect expired cert on load

### ISSUE THREAD

**Issue by agoddard** [sensu/sensu-go#4294]
## Feature Suggestion
For embedded etcd certificates, detect if cert is expired during backend start, and throw a fatal error if the certificate is expired (unless there's a corresponding flag to not care about cert expiry)

## Context
Right now, with an expired cert, users will see some x509 certificate date errors, but they will be hidden in a sea of etcd cluster connection errors which may trigger backend restarts, so the backend looks chaotic when it's just a simple cert issue

**Comment by portertech** [comment]
I see how this could be helpful on occasion, this would be considered feature work (validating certificate prior to passing it to the embedded etcd). I don't consider this feature work to be of a high priority. If we run into this case again, let's log it on this issue.

**Comment by agoddard** [comment]
agreed, we're going to work on docs for reference re: updating the self generated certs in the docs - but what I'm not clear on (and will also test) is whether an etcd cluster could potentially break if one of the member certs expires and the member is left out of the cluster for too long as a result.

**Comment by agoddard** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [Test self generated TLS cert expiry, document renewal](https://trello.com/c/rYh5DATc/579-test-self-generated-tls-cert-expiry-document-renewal)

**Comment by mgibson323** [comment]
After I completed testing with an expired cert for a single backend, I asked about the health of the etcd cluster in [Slack here](https://sumologic.slack.com/archives/C024XK35Z3Q/p1624395455002400)
